unittests/Commands/InterruptSignalAwaiterTests.cpp

```diff
@@ -0,0 +1,54 @@
+//===- unittests/Commands/InterruptSignalAwaiterTests.cpp -----------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "../lib/Commands/InterruptSignalAwaiter.h"
+
+#include "gtest/gtest.h"
+
+#include <chrono>
+#include <csignal>
+#include <thread>
+
+using namespace llbuild;
+using namespace llbuild::command;
+
+namespace {
+
+/// Check that we call the interrupt handler when raising a SIGINT signal.
+TEST(InterruptSignalAwaitertests, setInterruptHandler) {
+  bool calledInterruptHandler = false;
+  InterruptSignalAwaiter::GlobalAwaiter.setInterruptHandler([&] {
+    calledInterruptHandler = true;
+  });
+
+  std::raise(SIGINT);
+  
+  // Wait a bit to make sure the interrupt signal handler was called.
+  std::this_thread::sleep_for(std::chrono::nanoseconds(1));
```

Comment:
If we don't have this, for any length of time (1ns works), then the signal handler gets called after `EXPECT_TRUE(calledInterruptHandler);`